File: vimeosdk/src/main/java/gigigo/com/vimeolibs/VideoObserver.java
package gigigo.com.vimeolibs;


public final class VideoObserver extends DefaultObserver<VimeoInfo> {

  private final VimeoCallback getVideoCallback;

  public VideoObserver(VimeoCallback getVideoCallback) {
    this.getVideoCallback = getVideoCallback;
  }

  @Override public void onComplete() {

  }

  @Override public void onNext(VimeoInfo vimeoInfo) {
    if (getVideoCallback != null) {
      getVideoCallback.onSuccess(vimeoInfo);
    }
  }

  @Override public void onError(Throwable e) {
    if (getVideoCallback != null) {
      getVideoCallback.onError(new Exception(e));
    }
    e.printStackTrace();
  }
}

File: vimeosdk/src/main/java/gigigo/com/vimeolibs/VideoObserver.kt
package gigigo.com.vimeolibs

import android.util.Log
import io.reactivex.observers.DisposableObserver

class VideoObserver(
    private val getVideoCallback: VimeoCallback?) : DisposableObserver<VimeoInfo>() {
  override fun onComplete() {
  }

  override fun onNext(vimeoInfo: VimeoInfo) {
    getVideoCallback?.onSuccess(vimeoInfo)
  }

  override fun onError(e: Throwable) {
    getVideoCallback?.onError(Exception(e))
    Log.e(TAG, "VideoObserver", e)
  }

  companion object {
    private const val TAG = "VideoObserver"
  }
}